engine/resources/cron.go

```diff
@@ -462,17 +460,34 @@ func (obj *CronUID) IFF(uid engine.ResUID) bool {
 	if !ok {
 		return false
 	}
-	return obj.name == res.name
+	if obj.unit != res.unit {
+		return false
+	}
+	if obj.session != res.session {
+		return false
+	}
+	return true
+}
+
+// AutoEdges returns the AutoEdge interface.
+func (obj *CronRes) AutoEdges() (engine.AutoEdge, error) {
+	return nil, nil
 }
 
 // UIDs includes all params to make a unique identification of this object.
 // Most resources only return one although some resources can return multiple.
 func (obj *CronRes) UIDs() []engine.ResUID {
-	x := &CronUID{
+	unit := fmt.Sprintf("%s.service", obj.Name())
+	if obj.Unit != "" {
+		unit = obj.Unit
+	}
+	cronUID := &CronUID{
 		BaseUID: engine.BaseUID{Name: obj.Name(), Kind: obj.Kind()},
-		name:    obj.Name(), // svc name
+		unit:    unit,        // name of target unit
+		session: obj.Session, // user session
 	}
-	return append([]engine.ResUID{x}, obj.file.UIDs()...)
+	file, _ := obj.makeComposite() // error will be caught in validate
```

Comment:
To be safe, check the error, and instead of erroring, just *don't* append it... You need a *really* good excuse to not do something with an error, and this isn't it =D